### ISSUE TITLE
Asgiref 3.3.3 update breaks daphne 3.0.0

### ISSUE THREAD

**Issue by Ivo-Donchev** [django/daphne#363]
Hi,

We hit a problem an hour ago with our production daphne worker that we use for websockets connections after re-installing the dependancies. The error is:
```
Exception inside application: TypeError('async_to_sync can only be applied to async functions.')
``` 

After some research I saw that daphne has specified version of asgiref `>=3.2.10` - https://github.com/django/daphne/blob/main/setup.py#L25

And I saw that asgiref just updated to 3.3.3 https://github.com/django/asgiref/commit/07bffa4d017629ca472f67821a7984930aa6e08b

Our daphne version is `daphne==3.0.0`.

I believe that's a breaking issue since it'll breaks any daphne worker using 3.0.0 the next time it's deployed and install the dependencies. 

Thanks,
Ivo

**Comment by carltongibson** [comment]
Hi @Ivo-Donchev — thanks for the report. 

This looks like https://github.com/django/asgiref/issues/251 again. 

There's clearly some change that that changed the behaviour in an unexpected way, as we had a test failure on Django from it too.

**Comment by carltongibson** [comment]
@Ivo-Donchev It should be fixed in 3.3.3 — see https://github.com/django/asgiref/pull/252 — that should be out later today. 

(I'm assuming you meant 3.3.2...)

**Comment by carltongibson** [comment]
Ah no — I see this **is** 3.3.3 which is fresh out. 

Right OK...

**Comment by Ivo-Donchev** [comment]
Ah, yes, our breaking deploy was with `3.3.2`, sorry 

Great, I'm closing the issue as it's fixed in `3.3.3`. Thanks for the quick response @carltongibson
